app/controllers/admin/configs_controller.rb

```diff
@@ -212,15 +210,27 @@ def clean_up_params
       config[:credit_prices_in_cents]&.transform_values!(&:to_i)
     end
 
-    def toggle_email_password_authentication
-      if SiteConfig.allow_both_email_signup_and_login
-        SiteConfig.allow_email_password_registration = true
-        SiteConfig.allow_email_password_login = true
-      else
-        SiteConfig.allow_email_password_registration = false
-        SiteConfig.allow_email_password_login = false
-        SiteConfig.invite_only_mode = false
+    def disable_invite_mode_when_email_registration_off
+      return unless SiteConfig.allow_email_password_registration == false
+
+      SiteConfig.invite_only_mode = false
+    end
+
+    def update_enabled_auth_providers(value)
+      final_array = []
+      value.split(",").each do |entry|
+        final_array.push(entry) unless invalid_provider_entry(entry)
       end
+      SiteConfig.public_send("authentication_providers=", final_array) unless
+        prevent_all_auth_provider_disable?(final_array)
+    end
+
+    def invalid_provider_entry(entry)
+      entry.blank? || helpers.available_providers_array.exclude?(entry)
+    end
+
+    def prevent_all_auth_provider_disable?(value)
```

Comment:
Hey @joshpuetz since `SiteConfig.allow_email_password_login` will always be on, we do not need it in this check anymore, and can probably have `SiteConfig.public_send("authentication_providers=", final_array) unless value.empty?` on line 224